### PR TITLE
Add `Beatmap/Stack leniency` and reduce the corresponding part of `Client/Beatmap editor/Song Setup`

### PR DESCRIPTION
**Pull request by TicClick**
## Self-check

- [x] The changes are tested against the [contribution checklist](https://osu.ppy.sh/wiki/osu!_wiki/Contribution_guide#self-check)

### CODE REVIEW

File: wiki/Client/Beatmap_editor/Song_Setup/en.md [hunk 1]
```diff
@@ -238,19 +238,13 @@ You can, however, include the skin elements inside the beatmap folder. As most c
...
 
-In osu!, hit objects will automatically stack if they are both placed in the same spot while being close in-time. This is to help players see hit objects that overlap each other.
-
-Moving the slider towards the right will space out the "closeness" of hit objects in-time before stacking will occur. This option is best left as is to its default value of `7`.
-
-*Notice: Stacked hit circles in osu! will move towards the point where the stacking occurs.*
-
-Objects are considered to belong to same stack if they are at the same coordinate and no further in-time than `[(Approach Window) * (Stack Leniency) / 10]`, where `Approach Window` is the duration for which objects stay on screen (see [the section on approach rate](#approach-rate) above).
+The stack leniency value controls the maximum amount of time between objects, after which they won't be stacked. It is best left as is to its default value of `7`.
```
Comment (Walavouchey) [inline wiki/Client/Beatmap_editor/Song_Setup/en.md:247]:
```suggestion
The stack leniency value controls the maximum amount of time between objects, after which they will not be stacked. It is best left at its default value of `7`.
```

File: wiki/Client/File_formats/Osu_(file_format)/en.md [hunk 2]
```diff
@@ -31,7 +31,7 @@ The following content is separated into sections, indicated by section titles in
 | `PreviewTime` | Integer | Time in milliseconds when the audio preview should start | -1 |
 | `Countdown` | Integer | Speed of the countdown before the first hit object (`0` = no countdown, `1` = normal, `2` = half, `3` = double) | 1 |
 | `SampleSet` | String | Sample set that will be used if timing points do not override it (`Normal`, `Soft`, `Drum`) | Normal |
-| `StackLeniency` | Decimal | Multiplier for the threshold in time where hit objects placed close together stack (0–1) | 0.7 |
+| `StackLeniency` | Decimal | [Multiplier](/wiki/Beatmap/Stack_leniency) for the threshold in time where hit objects placed close together stack (0–1) | 0.7 |
```
Comment (Walavouchey) [inline wiki/Client/File_formats/Osu_(file_format)/en.md:34]: bit of a weird spot to put the link but, but i suppose it's due to the wording

File: wiki/Beatmap/Stack_leniency/en.md [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+  - autostacking
+---
+
+# Stack leniency
+
+**Stack leniency** is a parameter of a [beatmap](/wiki/Beatmap) which controls the formation of automatic [stacks](/wiki/Beatmapping/Mapping_techniques/Stack) of [circles](/wiki/Gameplay/Hit_object/Hit_circle) and [sliders](/wiki/Gameplay/Hit_object/Slider) in the [osu!](/wiki/Game_mode/osu!) game mode. Lower stack leniency reduces the maximum amount of time after which [hit objects](/wiki/Gameplay/Hit_object) are no longer placed in the same stack; likewise, higher stack leniency will group together objects that are placed further in time.
+
+## Description
+
+Stack leniency may be changed from the `Advanced` tab of the [Song Setup](/wiki/Client/Beatmap_editor/Song_Setup) section of the beatmap editor. While its displayed possible values in the editor range from `0` to `10`, they map to the `0.0`–`1.0` range of the `StackLeniency` parameter in [`.osu` files](/wiki/Client/File_formats/Osu_(file_format)).
```
Comment (Walavouchey) [inline wiki/Beatmap/Stack_leniency/en.md:21]:
```suggestion
Stack leniency may be changed from the `Advanced` tab of the [Song Setup](/wiki/Client/Beatmap_editor/Song_Setup) section of the [beatmap editor](/wiki/Client/Beatmap_editor). While its displayed possible values in the editor range from `0` to `10`, they map to the `0.0`–`1.0` range of the `StackLeniency` parameter in [`.osu` files](/wiki/Client/File_formats/Osu_(file_format)).
```

File: wiki/Beatmap/Stack_leniency/en.md [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+
+![Stacking hit objects while editing may make them overlap to be on top of each other.](img/stacking-editor.jpg "Stacking hit objects while editing")
+
+![However, while playing, the game will slope them for the player.](img/stacking-playtest.jpg "Stacked hit objects while playing")
+
+The stack leniency multiplier, together with the [approach rate](/wiki/Beatmap/Approach_rate) value, controls which circles and slider parts are considered for stacking.[^stacking-algorithm] The stack is composed of the objects which are no longer than `preempt * StackLeniency` milliseconds apart, where `preempt` is the [approach window width](/wiki/Beatmap/Approach_rate#animation-timing), and `StackLeniency` is the value taken from the beatmap's `.osu` file.
+
+The minimal stack leniency value, `0`, disables automatic stacks completely, while at its maximal value, `1`, objects are considered stacked from the moment they start becoming visible.
+
+## Notes and references
```
Comment (Walavouchey) [inline wiki/Beatmap/Stack_leniency/en.md:33]:
```suggestion
## References
```

File: wiki/Beatmap/Stack_leniency/en.md [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+
+[^stacking-algorithm]: [A code snippet by peppy (2011-08-24) "osu! stacking algorithm"](https://gist.github.com/peppy/1167470)
```
Comment (Walavouchey) [inline wiki/Beatmap/Stack_leniency/en.md:35]:
```suggestion
[^stacking-algorithm]: [Code snippet by peppy (2011-08-24) "osu! stacking algorithm"](https://gist.github.com/peppy/1167470)
```

File: wiki/Beatmap/Stack_leniency/en.md [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+
+Stack leniency may be changed from the `Advanced` tab of the [Song Setup](/wiki/Client/Beatmap_editor/Song_Setup) section of the beatmap editor. While its displayed possible values in the editor range from `0` to `10`, they map to the `0.0`–`1.0` range of the `StackLeniency` parameter in [`.osu` files](/wiki/Client/File_formats/Osu_(file_format)).
+
+## Behaviour
+
+![Stacking hit objects while editing may make them overlap to be on top of each other.](img/stacking-editor.jpg "Stacking hit objects while editing")
+
+![However, while playing, the game will slope them for the player.](img/stacking-playtest.jpg "Stacked hit objects while playing")
+
+The stack leniency multiplier, together with the [approach rate](/wiki/Beatmap/Approach_rate) value, controls which circles and slider parts are considered for stacking.[^stacking-algorithm] The stack is composed of the objects which are no longer than `preempt * StackLeniency` milliseconds apart, where `preempt` is the [approach window width](/wiki/Beatmap/Approach_rate#animation-timing), and `StackLeniency` is the value taken from the beatmap's `.osu` file.
```
Comment (Walavouchey) [inline wiki/Beatmap/Stack_leniency/en.md:29]:
```suggestion
Stack leniency, together with the [approach rate](/wiki/Beatmap/Approach_rate), controls which circles and slider parts are considered for stacking.[^stacking-algorithm] The stack is composed of the objects which are no longer than `preempt * StackLeniency` milliseconds apart, where `preempt` is the [approach window width](/wiki/Beatmap/Approach_rate#animation-timing), and `StackLeniency` is the value taken from the beatmap's `.osu` file.
```

File: wiki/Beatmap/Stack_leniency/en.md [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+  - automatic stack
+  - automated stack
+  - automatic stacking
+  - automated stacking
+  - autostacking
+---
+
+# Stack leniency
+
+**Stack leniency** is a parameter of a [beatmap](/wiki/Beatmap) which controls the formation of automatic [stacks](/wiki/Beatmapping/Mapping_techniques/Stack) of [circles](/wiki/Gameplay/Hit_object/Hit_circle) and [sliders](/wiki/Gameplay/Hit_object/Slider) in the [osu!](/wiki/Game_mode/osu!) game mode. Lower stack leniency reduces the maximum amount of time after which [hit objects](/wiki/Gameplay/Hit_object) are no longer placed in the same stack; likewise, higher stack leniency will group together objects that are placed further in time.
```
Comment (Walavouchey) [inline wiki/Beatmap/Stack_leniency/en.md:17]:
```suggestion
**Stack leniency** is a parameter of a [beatmap](/wiki/Beatmap) which controls the formation of automatic [stacks](/wiki/Beatmapping/Mapping_techniques/Stack) of [circles](/wiki/Gameplay/Hit_object/Hit_circle) and [sliders](/wiki/Gameplay/Hit_object/Slider) in the [osu!](/wiki/Game_mode/osu!) game mode. A lower stack leniency reduces the maximum amount of time after which [hit objects](/wiki/Gameplay/Hit_object) are no longer placed in the same stack; likewise, higher stack leniency will group together objects that are placed further in time.
```

**Review by Walavouchey — CHANGES_REQUESTED** [review]

File: wiki/Client/File_formats/Osu_(file_format)/en.md (hunk 2 above)
Comment (TicClick) [reply wiki/Client/File_formats/Osu_(file_format)/en.md:34]: yeah

File: wiki/Beatmap/Stack_leniency/en.md (hunk 4 above)
Comment (TicClick) [reply wiki/Beatmap/Stack_leniency/en.md:33]: whoops it becomes a habit

**Review by Walavouchey — APPROVED** [review]

**Pull request merged**